projects/Mallard/src/helpers/push-tracking.ts

```diff
@@ -0,0 +1,53 @@
+import AsyncStorage from '@react-native-community/async-storage'
+import { sendComponentEvent, ComponentType, Action } from 'src/services/ophan'
+import { londonTime } from 'src/helpers/date'
+import { errorService } from 'src/services/errors'
+
+const PUSH_TRACKING_KEY = '@push-tracking'
+
+interface Tracking {
+    time: string
+    id: string
+    value: string
+}
+
+const getPushTracking = async (): Promise<string | null> =>
+    AsyncStorage.getItem(PUSH_TRACKING_KEY)
+
+const clearPushTracking = async (): Promise<void> =>
+    AsyncStorage.removeItem(PUSH_TRACKING_KEY)
+
+const pushTracking = async (id: string, value: string) => {
+    sendComponentEvent({
+        componentType: ComponentType.appVideo,
+        action: Action.view,
+        value,
+        componentId: id,
+    })
+
+    try {
+        const storedTracking = await AsyncStorage.getItem(PUSH_TRACKING_KEY)
+        const tracking: Tracking = {
+            time: londonTime().format(),
+            id,
+            value,
+        }
+
+        const saveTracking = storedTracking
+            ? [...JSON.parse(storedTracking), tracking]
+            : [tracking]
+
+        try {
+            await AsyncStorage.setItem(
+                PUSH_TRACKING_KEY,
+                JSON.stringify(saveTracking),
+            )
+        } catch (e) {
```

Comment:
Teeny one but we can probable ditch the inner `catch` here as it looks redundant.

Reply:
of course - it will bubble up.